src/client/components/Header.tsx

```diff
@@ -39,10 +61,43 @@ export const Header: React.FunctionComponent = (props) => {
             ? null :
               isLoggedIn
                 ? [
-                    <IconButton color="inherit" key={0}><SettingsIcon color="inherit"/></IconButton>,
+                    <IconButton color="inherit" key={0} onClick={() => setOpen(true)}><SettingsIcon color="inherit" /></IconButton>,
                     <Fragment key={1}>{data.me.name}</Fragment>,
-                    <Button onClick={logOutUser} key={2}>Log Out</Button>
+                    <Button onClick={logOutUser} key={2}>Log Out</Button>,
+                    <Modal open={open} style={{
+                      display: 'flex',
+                      alignItems: 'center',
+                      justifyContent: 'center',
+                    }}>
+                      <Paper style={{
+                        minWidth: "500px",
+                        maxWidth: "800px",
+                        margin: "auto"
+                        }}>
+                        <Grid container direction="column" alignContent="center" alignItems="center" spacing={1}>
+                          <Grid xs={12} item>
+                            Name: {data.me.name}
+                          </Grid>
+                          <Grid xs={12} item>
+                            Email: {data.me.email}
+                          </Grid>
+                            Edit Password:
+                            <TextField id="old-password" label="Enter Old Password" autoFocus value={oldPassword} onChange={onTextFieldUpdate(setOldPassword)}></TextField>
```

Comment:
You can only have one text field with `autoFocus` which should usually be the first item in the form, so get rid of the autoFocus on the other 2 text fields.